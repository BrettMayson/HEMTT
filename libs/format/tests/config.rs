use hemtt_format::{FormatterConfig, format_config};

const ROOT: &str = "tests/config/";

macro_rules! config {
    ($dir:ident) => {
        paste::paste! {
            #[test]
            fn [<config_ $dir>]() {
                insta::assert_snapshot!(config(stringify!($dir)));
            }
        }
    };
}

config!(array);
config!(basic);
config!(classes);
config!(comments);
config!(eject);
config!(empty);
config!(include);
config!(invalid);
config!(macro);
config!(macro_path);
config!(math);
config!(nested);
config!(numbers);
config!(parent);

fn config(file: &str) -> String {
    format_config(
        &std::fs::read_to_string(format!("{ROOT}{file}.hpp")).expect("Failed to read test file"),
        &FormatterConfig::default(),
    )
    .unwrap_or_else(|err| format!("ERROR: {err}"))
}
