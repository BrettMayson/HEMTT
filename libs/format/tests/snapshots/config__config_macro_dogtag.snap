---
source: libs/format/tests/config.rs
expression: config(stringify! (macro_dogtag))
---
#define ACE_DOGTAG_ITEM(DOGTAGBASE,DOGTAGID) \
    class DOGTAGBASE##_##DOGTAGID: DOGTAGBASE { \
        author = ECSTRING(common,ACETeam); \
        scope = 1; \
        scopeArsenal = 0; \
        scopeCurator = 0; \
        descriptionShort = CSTRING(itemName); \
        GVAR(tagID) = DOGTAGID; \
    }

class CfgWeapons {
    class ACE_ItemCore;
    class CBA_MiscItem_ItemInfo;

    class ACE_dogtag: ACE_ItemCore {
        author = ECSTRING(common,ACETeam);
        scope = 0;
        displayName = CSTRING(itemName);
        model = QPATHTOF(data\ace_dogtag.p3d);
        picture = QPATHTOF(data\dogtagSingle.paa);
        class ItemInfo: CBA_MiscItem_ItemInfo {
            mass = 0; //too small to for 1 ?
        };
    };
    ACE_DOGTAG_ITEM(ACE_dogtag,1);
    ACE_DOGTAG_ITEM(ACE_dogtag,2);
    ACE_DOGTAG_ITEM(ACE_dogtag,3);
    ACE_DOGTAG_ITEM(ACE_dogtag,4);
    ACE_DOGTAG_ITEM(ACE_dogtag,5);
    ACE_DOGTAG_ITEM(ACE_dogtag,6);
    ACE_DOGTAG_ITEM(ACE_dogtag,7);
    ACE_DOGTAG_ITEM(ACE_dogtag,8);
    ACE_DOGTAG_ITEM(ACE_dogtag,9);
    ACE_DOGTAG_ITEM(ACE_dogtag,10);
    ACE_DOGTAG_ITEM(ACE_dogtag,11);
    ACE_DOGTAG_ITEM(ACE_dogtag,12);
    ACE_DOGTAG_ITEM(ACE_dogtag,13);
    ACE_DOGTAG_ITEM(ACE_dogtag,14);
};
